services/common/snmp/ntcip1218/rsu_firmware_version.py

```diff
@@ -0,0 +1,44 @@
+import subprocess
+import logging
+import common.snmp.snmpcredential as snmpcredential
+import common.snmp.snmperrorcheck as snmperrorcheck
+
+
+def get(rsu_ip, snmp_creds):
+    """
+    Retrieves the RSU's firmware version using SNMP NTCIP-1218 OID.
+    Returns a dictionary with the firmware version as a string.
+    """
+    snmpget_result = None
+
+    try:
+        output = ""
+
+        # Create the SNMPGet command
+        cmd = "snmpget -v 3 -t 5 {auth} {rsuip} NTCIP1218-v01::rsuFirmwareVersion.0".format(
+            auth=snmpcredential.get_authstring(snmp_creds), rsuip=rsu_ip
+        )
+
+        # Example console output for a functional RSU
+        # NTCIP1218-v01::rsuFirmwareVersion.0 = STRING: y20.61.1-b275956
+        logging.info(f"Running snmpget: {cmd}")
+        output = subprocess.run(cmd, shell=True, capture_output=True, check=True)
+        output = output.stdout.decode("utf-8").split("\n")[:-1]
+    except subprocess.CalledProcessError as e:
+        output = e.stderr.decode("utf-8").split("\n")[:-1]
+        logging.error(
+            f"Encountered error while running snmpget NTCIP1218-v01::rsuFirmwareVersion: {output[-1]}"
+        )
+        err_message = snmperrorcheck.check_error_type(output[-1])
+        return {"RsuStatus": err_message}, 500
+
+    if len(output) == 1:
```

Comment:
This logic is a bit confusing to me, we check to see there's only one returned value in 'output', then iterate over the list to get the snmpget_result. Is it possible that we'd have more than one value in output? If so, we'd see 'none' returned from this method, is that acceptable? Should we be logging it if we run into that?